cmd/injective-price-oracle/oracle.go

```diff
@@ -216,6 +250,63 @@ func oracleCmd(cmd *cli.Cmd) {
 			svc.Close()
 		})
 
+		go func() {
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			// Channel to signal when an error occurs and a retry is needed
+			retryChan := make(chan struct{}, 1)
+
+			// Function to connect to the WebSocket and handle errors
+			reconnect := func() error {
+				conn, err := pipeline.ConnectWebSocket(ctx, *websocketUrl, *websocketHeader, oracle.MaxRetriesReConnectWebSocket)
+				if err != nil {
+					return err
+				}
+
+				if err := storkFetcher.Reconnect(conn); err != nil {
+					if errors.Is(err, oracle.ErrInvalidMessage) {
+						log.Errorln("invalid message received from WebSocket")
+					}
+
+					log.WithError(err).Errorln("failed to connect to WebSocket")
+					return err
+				}
+
+				return nil
+			}
+
+			// Start initial connection attempt
+			if err := storkFetcher.Start(); err != nil {
+				// If there's an error, trigger a retry
+				retryChan <- struct{}{}
+			}
+
+			for {
```

Comment:
all the initi and retry is a bit convoluted imo, if start receives the context and the connection in the Start method we can simplify the logic a lot

Something like this should be enough
```go
go func() {
	if storkFetcher == nil {
		return // no stork feeds
	}
	connectIn := 0 * time.Second
	for {
		select {
		case <-ctx.Done():
			return
		case <-time.After(connectIn):
		}

		connectIn = 5 * time.Second
		conn, err := pipeline.ConnectWebSocket(ctx, *websocketUrl, *websocketHeader, oracle.MaxRetriesReConnectWebSocket)
		if err != nil {
			log.WithError(err).Errorln("failed to connect to WebSocket")
			continue
		}

		err := storkFetcher.Start(ctx, conn)
		if err != nil {
			log.WithError(err).Errorln("stork fetcher failed")
		}
	}
}()
```